src/Control/Director.php

```diff
@@ -704,7 +704,7 @@ public static function baseFolder()
      */
     public static function publicDir()
     {
-        $alternate = self::config()->uninherited('alternate_public_dir');
+        $alternate = Deprecation::withNoReplacement(fn() => self::config()->uninherited('alternate_public_dir'));
```

Comment:
Please use the standard non-short function() here that we used for everything else

I know this works, however if we need to Ctrl-F to find everything later for whatever reason this make it harder to find

Reply:
"standard" here is subjective. Short arrow function syntax was only introduced in PHP 7.4, so we _couldn't_ use this until [4.11.0](https://docs.silverstripe.org/en/4/changelogs/4.11.0/#phpeol) which dropped support for PHP 7.3 and that was just in June last year. I'd say the only reason using a full `function () { }` syntax for these is "standard" is because we haven't been able to use the new syntax until very recently.
I would prefer to keep this new syntax here - in my mind "we haven't used this syntax before" is a really bad reason to _not_ use new syntax, given the only reason we hadn't used it before it we literally couldn't do that without breaking things.

Reply:
If you're going to insist on this change, we probably need some more opinions from the core committers about whether this will be a rule going forward - and if it is, we should add linting to catch it.

Reply:
In this case, I'd keep it "the old way" and for CMS5 we can do it the new way.
Re: the search argument you can find either, especially when these closures aren't that common.

Then as a bonus, we can create a few MNT cards for using things like `str_contains` instead of `strpos !== false`, null safe object operator `?->` etc. Lead the way with modern code.

Reply:
All I'm saying is make it match the syntax of the other several dozen other withNoReplacement replacement calls so things are consistent. 

`Deprecation::withNoReplacement(function () {` won't match it which is what I'd personally be liable to copy paste

Reply:
Since there's two of you saying to use the old way for this, and one of you isn't gonna merge until I do, I'll make that change.
I do think we should start using new syntax though - be forewarned that for things like this I'm going to start pushing more and more for the new syntax to be used.

Having cards to update some of the syntax as well is a good idea - but lets look at that sort of thing once we've got 5 stable released and have some more time for that sort of enhancement.

Reply:
Anyway I've made the change now